core-contracts/FeeHandler/src/main/java/network/balanced/score/core/feehandler/FeeHandlerImpl.java

```diff
@@ -203,6 +262,17 @@ public void tokenFallback(Address _from, BigInteger _value, byte[] _data) {
                 transferToken(sender, getContractAddress("dividends"), getTokenBalance(sender), new byte[0]);
             }
         }
+        if (_from == loans.get()) {
```

Comment:
For all these ifs, use if, else if to save some gas. Also change compares to use .equals. 
```suggestion
        if (_from.equals(loans.get())) {
```

Reply:
Fixed for all address comparisons.